### PR TITLE
Inject neutral plasma in checkpoint restart tests

### PR DESCRIPTION
**Pull request by KZhu-ME**
Added neutral plasma injection to checkpoint restart tests.
Also:
 - added a check that net charge density is the same for the restarted run compared with the original run.

### CODE REVIEW

File: mewarpx/tests/test_checkpoint_and_restart.py [hunk 1]
```diff
@@ -223,3 +240,8 @@ def test_extra_steps_after_restart():
     end_step = mwxrun.get_it()
 
     assert start_step + additional_steps == end_step
+
+    restart_net_charge_density = np.load("diags/fields/Net_charge_density_0000000006.npy")
+    original_net_charge_density = np.load("../../test_files/checkpoint/Net_charge_density_0000000006.npy")
```
Comment (roelof-groenewald) [inline mewarpx/tests/test_checkpoint_and_restart.py]:
```suggestion
    original_net_charge_density = np.load(os.path.join(
        testing_util.test_dir, "checkpoint",
        "Net_charge_density_0000000006.npy"
    ))
```

**Review by roelof-groenewald — APPROVED** [review]
Just one change suggested, otherwise looks good!

**Review by peterscherpelz — APPROVED** [review]
+1 what Roelof said. Thanks!

File: mewarpx/tests/test_checkpoint_and_restart.py (hunk 1 above)
Comment (KZhu-ME) [reply mewarpx/tests/test_checkpoint_and_restart.py]: Good catch, I changed all the other file paths to use `os.path.join` as well.

**Pull request merged**